arcgis-ios-sdk-samples/Display information/Create symbol styles from web styles/CreateSymbolStylesFromWebStylesViewController.swift

```diff
@@ -0,0 +1,306 @@
+            // Hold a weak reference to the table view to asynchronously reload
+            // image swatches for the symbols.
+            symbolsDataSource.tableView = controller.tableView
+        }
+    }
+    
+    func adaptivePresentationStyle(for controller: UIPresentationController, traitCollection: UITraitCollection) -> UIModalPresentationStyle {
+        // Ensure that the settings are shown in a popover on small displays.
+        return .none
+    }
+}
+
+// MARK: - SymbolsDataSource
+
+private class SymbolsDataSource: NSObject {
+    /// A weak reference to the table view which the data source itself connects.
+    weak var tableView: UITableView?
+    
+    /// The symbols created from a symbol style.
+    private var symbols = [(category: SymbolCategory, symbol: AGSSymbol)]() {
+        didSet {
+            // Keep the array sorted whenever it is modified.
+            symbols.sort { $0.category < $1.category }
+            // Reset caches to ensure legends are updated correctly.
+            resetCaches()
+        }
+    }
+    
+    /// A cache for the image swatches of the symbols.
+    private var cachedImages = [IndexPath: UIImage]()
+    /// A cache for cancelable operations.
+    private var imageOperations = [IndexPath: AGSCancelable]()
+    
+    private func resetCaches() {
+        imageOperations.forEach { _, operation in
+            operation.cancel()
+        }
+        imageOperations.removeAll()
+        cachedImages.removeAll()
+    }
+    
+    /// Get certain categories of symbols from a symbol style.
+    /// - Parameters:
+    ///   - symbolStyle: An `AGSSymbolStyle` object.
+    ///   - categories: The types of symbols to search in the symbol style.
+    ///   - completion: A closure executed upon success.
+    func getSymbols(symbolStyle: AGSSymbolStyle, categories: [SymbolCategory], completion: @escaping () -> Void) {
+        let getSymbolsGroup = DispatchGroup()
+        var symbols = [(SymbolCategory, AGSSymbol)]()
+        
+        categories.forEach { category in
+            getSymbolsGroup.enter()
+            symbolStyle.symbol(forKeys: [category.symbolName]) { symbol, _ in
+                defer {
+                    getSymbolsGroup.leave()
+                }
+                // Add the symbol to the result collection and ignore any error.
+                if let symbol = symbol {
+                    symbols.append((category, symbol))
+                }
+            }
+        }
+        getSymbolsGroup.notify(queue: .main) { [weak self] in
+            self?.symbols = symbols
+            completion()
+        }
+    }
+    
+    /// Create an `AGSUniqueValueRenderer` to render feature layer with symbol styles.
+    /// - Parameter fieldNames: The attributes to match the unique values against.
+    /// - Returns: An `AGSUniqueValueRenderer` object.
+    func makeUniqueValueRenderer(fieldNames: [String]) -> AGSUniqueValueRenderer {
```

Comment:
The data source seems like a really odd place for this function as it has nothing to do with the table view. I would suggest moving it back to the sample view controller.